### PR TITLE
Add basic support for yubikey otp 2fa

### PR DESCRIPTION
**Pull request by hdiniz**
Related https://github.com/opencollective/opencollective/issues/5294
Requires https://github.com/opencollective/opencollective-api/pull/8668

# Description

Add support to input YubiKey OTP for 2FA. Initially will only be enabled for internal users, modal will need better design work before general release. 

# Screenshots

image

### CODE REVIEW

File: components/two-factor-authentication/TwoFactorAuthenticationModal.js [hunk 1]
```diff
@@ -50,34 +77,70 @@ export default function TwoFactorAuthenticationModal() {
...
+      defaultMessage="Please enter your 6-digit code without any dashes or select the input below, plug your YubiKey and press it to generate a code."
+    />
+  ) : (
+    <FormattedMessage
+      id="TwoFactorAuth.Setup.Form.InputLabel"
+      defaultMessage="Please enter your 6-digit code without any dashes."
+    />
+  );
+
+  const placeholder = isUsingRecoveryCode ? '' : hasYubikeyOTP ? '123456 or YubiKey: cccc...' : '123456';
```
Comment (Betree) [inline components/two-factor-authentication/TwoFactorAuthenticationModal.js:113]: Good call adapting the label depending on `supportedMethods` :+1:

File: components/SignInOrJoinFree.js [hunk 2]
```diff
@@ -235,144 +215,10 @@ class SignInOrJoinFree extends React.Component {
     }
   };
 
-  renderTwoFactorAuthBoxes = useRecoveryCodes => {
```
Comment (Betree) [inline components/SignInOrJoinFree.js:238]: Good job moving that out of the component

**Review by Betree — APPROVED** [review]
Looking good!

**Comment by hdiniz** [comment]
@Betree a few more changes were added here to support dynamic methods returned by the signin api

**Pull request merged**